frontend/src/layout/navigation/navigationLogic.ts

```diff
@@ -152,8 +152,18 @@ export const navigationLogic = kea<navigationLogicType<WarningType>>({
             null as string | null,
             {
                 loadLatestVersion: async () => {
-                    const versions = await api.get('https://update.posthog.com/versions')
-                    return versions[0].version
+                    let retries = 0
```

Comment:
I think this can be simplified significantly. Not 100% sure `breakpoint` is available - if not, your code will do the trick.

```js
loadLatestVersion: async (_, breakpoint) => {
  for (let i = 0; i < 3; i++) {
    const versions = await api.get('https://update.posthog.com/versions')
    if (versions && versions.length > 0) {
      await breakpoint(150)
    }
  }
}
}
return null
```